### DISCUSSION TITLE
How to create tables in order

### DISCUSSION THREAD

**Discussion by lthoa** [oqtane/oqtane.framework#1839] [Q&A]
I create a number of modules, each module has a few tables in the database, the tables in this module are related to the tables in other modules, so creating tables must be done in a certain order, otherwise there will be problems. relationship error between tables.
How to make the above request when setting up the website?

Thanks?

**Comment by sbwalker** [comment]
If you expect that the modules will always be used together then one option is to put all of your migration logic in one of the modules. Another somewhat related option would be to consolidate your various module projects into a single project ( ie. there is no rule that modules must be in their own project/assembly ).

**Reply by lthoa** [reply]
I think option 1 is good for large projects, option 2 is suitable for small projects.

Thanks @sbwalker!
